### PR TITLE
Add Generator class

### PR DESCRIPTION
**Pull request by tvararu**
This will be the class responsible for the main business logic, which is combining country/territory/uk Register data with a synonym CSV into a JSON picker data file.

Currently it only covers the most basic use case, which is passing in the country register (as JSON) and getting back a resulting picker graph. It doesn't handle territories or synonyms yet.

To help with testing I've deployed it to AWS Lambda. You can try it out by sending a POST request to https://ic0bz9k68j.execute-api.us-east-1.amazonaws.com/dev/generate:

```
POST /dev/generate HTTP/1.1
Host: ic0bz9k68j.execute-api.us-east-1.amazonaws.com
Content-Type: application/json
Cache-Control: no-cache

{
  "GB": {
    "index-entry-number": "6",
    "entry-number": "6",
    "entry-timestamp": "2016-04-05T13:23:05Z",
    "key": "GB",
    "item": [
      {
        "country": "GB",
        "official-name": "The United Kingdom of Great Britain and Northern Ireland",
        "name": "United Kingdom",
        "citizen-names": "Briton;British citizen"
      }
    ]
  }
}
```

Expected response:

```
{
    "country:GB": {
        "names": {
            "en-GB": "United Kingdom"
        },
        "meta": {
            "canonical": true,
            "canonical-mask": 1,
            "stable-name": true
        },
        "edges": {
            "from": []
        }
    }
}
```

### CODE REVIEW

File: src/test/java/uk/gov/GeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+				+ "\"meta\": {"
+					+ "\"canonical\": true,"
+					+ "\"canonical-mask\": 1,"
+					+ "\"stable-name\": true"
+				+ "},"
+				+ "\"edges\": {"
+					+ "\"from\": []"
+				+ "}"
+			+ "}"
+		+ "}";
```
Comment (tvararu) [inline src/test/java/uk/gov/GeneratorTest.java]: Is this a sane way to write JSON strings in Java?

File: src/test/java/uk/gov/GeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+	@Test
+	public void run() {
+		Generator generator = new Generator();
+		String output = generator.run(this.countryJSON, this.csv);
```
Comment (tvararu) [inline src/test/java/uk/gov/GeneratorTest.java]: Is this a good pattern for the module? Instantiate it, and then pass in arguments to a method to get the results? Or is it better to construct the instance with the arguments and then have a getter for the result?

My gut is to keep the class as stateless as possible and rely on the consumer to keep track of state.

File: src/test/java/uk/gov/GeneratorTest.java (hunk 1 above)
Comment (tvararu) [reply src/test/java/uk/gov/GeneratorTest.java]: Also, should these be refactored as fixtures in a separate file?

**Review by michaelabenyohai — COMMENTED** [review]
👍

**Pull request merged**